geometric/engine.py

```diff
@@ -379,6 +384,48 @@ def calc_new(self, coords, dirname):
             raise OpenMMEngineError
         return energy, gradient
 
+
+    @staticmethod
+    def opls(system):
+        """Apply the opls combination rule to the system."""
+
+        from numpy import sqrt
+        import simtk.openmm as mm
+
+        # get system information from the openmm system
+        forces = {system.getForce(index).__class__.__name__: system.getForce(index) for index in
+                  range(system.getNumForces())}
+        # use the nondonded_force tp get the same rules
+        nonbonded_force = forces['NonbondedForce']
+        lorentz = mm.CustomNonbondedForce(
+            'epsilon*((sigma/r)^12-(sigma/r)^6); sigma=sqrt(sigma1*sigma2); epsilon=sqrt(epsilon1*epsilon2)*4.0')
+        lorentz.setNonbondedMethod(mm.CustomNonbondedForce.NoCutoff)
+        lorentz.addPerParticleParameter('sigma')
+        lorentz.addPerParticleParameter('epsilon')
+        lorentz.setCutoffDistance(nonbonded_force.getCutoffDistance())
+        system.addForce(lorentz)
+        ljset = {}
+        # Now for each particle calculate the combination list again
+        for index in range(nonbonded_force.getNumParticles()):
+            charge, sigma, epsilon = nonbonded_force.getParticleParameters(index)
+            # print(nonbonded_force.getParticleParameters(index))
+            ljset[index] = (sigma, epsilon)
+            lorentz.addParticle([sigma, epsilon])
+            nonbonded_force.setParticleParameters(
+                index, charge, 0, 0)
+        for i in range(nonbonded_force.getNumExceptions()):
+            (p1, p2, q, sig, eps) = nonbonded_force.getExceptionParameters(i)
+            # ALL THE 12,13 and 14 interactions are EXCLUDED FROM CUSTOM NONBONDED
```

Comment:
Does this ignore the typical 0.5 scaling factor using for 1-4 interactions?

Reply:
No, it is used we state it in the FF XML file and are just changing how the sigma values are combined here, I think this comment needs updating.

Reply:
No problem I think ill change this and resubmit the pull request?